### PR TITLE
(HTCONDOR-2615)  Implement `htcondor2.set_ready_state()`.

### PR DESCRIPTION
**Pull request by Todd-L-Miller**
The credmons require this to function, and the regression tests use the credmons.

We run one test with the dummy oauth credmon, and two tests with the real oauth credmon; those two credmons have been updated.  The other credmons (and some utility code) have not been changed, because they're aren't being tested, and it would be irresponsible as such.

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

**Review by JaimeFrey — CHANGES_REQUESTED** [review]
This is not how you get the family security session. It's passed in a separate environment variable (CONDOR_PRIVATE_INHERIT). See get_family_session() in src/python-bindings/dc_tool.cpp.

**Comment by Todd-L-Miller** [comment]
Yeah.  I honestly don't know what I was thinking there, or why it pretended to work,  Any thoughts on `getSecMan()`?

File: src/condor_daemon_client/daemon.h [hunk 1]
```diff
@@ -672,6 +672,9 @@ class Daemon: public ClassyCountedPtr {
 	void setSecSessionId(const std::string& sess_id) { m_sec_session_id = sess_id; }
 	const std::string& getSecSessionId() { return m_sec_session_id; }
 
+    // FIXME: ask JaimeF about this.
+    SecMan & getSecMan() { return _sec_man; }
+
```
[the remark this answers is not in the document]
Comment (Todd-L-Miller) [reply src/condor_daemon_client/daemon.h]: Not now that we've given up on family security, anyway.  Removed.

**Review by JaimeFrey — APPROVED** [review]
LGTM

**Pull request merged**